src/scenes/MembershipMintPage/PartnerMembershipMintPage.tsx

```diff
@@ -0,0 +1,178 @@
+  '0xb3d5858d11ff402e3626dba53009cb46666f3c54': 'Gallery Membership Card',
+};
+
+function encodeParameters(parameters: Array<string | number>) {
+  return web3.eth.abi.encodeParameters(
+    ['address', 'uint256', 'uint256', 'uint256', 'uint256'],
+    parameters
+  );
+}
+
+// reduces a list of Opensea Assets to a list just the contract addresses
+export function reduceOpenseaAssetsToContractAddresses(assets: OpenseaAsset[]) {
+  const partnerNftAddressesOwned = assets.reduce(
+    (acc: Record<string, boolean>, asset: OpenseaAsset) => {
+      if (asset.asset_contract.address) {
+        acc[asset.asset_contract.address] = true;
+      }
+
+      return acc;
+    },
+    {}
+  );
+  return Object.keys(partnerNftAddressesOwned);
+}
+
+function generateMerkleProof(encodedAbi1: string, encodedAbi2: string) {
+  const elements = [encodedAbi1, encodedAbi2];
+  const merkleTree = new MerkleTree(elements);
+  return merkleTree.getHexProof(elements[0]);
+}
+
+// Checks to see if the address owns any partner NFTs by
+// retrieving all NFTs owned by a user from opensea,
+// and filtering the list by whitelisted partner contract addresses
+async function detectOwnedPartnerNftsFromOpensea(
+  account: string,
+  addresses: string[]
+): Promise<string[]> {
+  const params = getContractAddressQueryParams(addresses);
+  const response = await fetch(
+    `https://testnets-api.opensea.io/api/v1/assets?owner=${account}${params}`,
+    {}
+  );
+
+  const responseBody = await response.json();
+  // const responseBody = { assets: [] };
+  if (responseBody.assets.length > 0) {
+    return reduceOpenseaAssetsToContractAddresses(responseBody.assets);
+  }
+
+  return [];
+}
+
+async function detectOwnedGeneralCardsFromOpensea(account: string) {
+  const response = await fetch(
+    `https://testnets-api.opensea.io/api/v1/assets?owner=${account}&asset_contract_addresses=${GENERAL_MEMBERSHIP_CONRTACT_ADDRESS}`,
+    {}
+  );
+
+  const responseBody = await response.json();
+  return responseBody.assets.length > 0;
+}
+
+// Turns list of contract addresses into url parameters for the GET call to Opensea
+function getContractAddressQueryParams(addresses: string[]) {
+  return addresses.map((address) => `&asset_contract_addresses=${address}`).join('');
+}
+
+function PartnerMembershipMintPageContent() {
+  const { account } = useWeb3React<Web3Provider>();
+  const contract = useGeneralMembershipCardContract();
+
+  const { getSupply } = useMembershipMintPageActions();
+
+  const [ownedPartnerNfts, setOwnedPartnerNfts] = useState<string[]>([]);
+  const [ownsGeneralCard, setOwnsGeneralCard] = useState(false);
+  const canMintToken = useMemo(
+    () => !ownsGeneralCard && ownedPartnerNfts.length > 0,
+    [ownedPartnerNfts.length, ownsGeneralCard]
+  );
+
+  // Regardless of how many partner NFTs they own, we just need one contract address to mint with, so pick first one in list.
+  const partnerContractAddress = useMemo(() => ownedPartnerNfts[0], [ownedPartnerNfts]);
```

Comment:
Nit: this line and the one above we don't have to memo since they're constant time computations- memo is more useful if we're filtering an array or finding for example

Reply:
true, done